server/server.go

```diff
@@ -206,7 +146,38 @@ func (server *GripServer) Serve(pctx context.Context) error {
 			if server.conf.DisableHTTPCache {
 				resp.Header().Set("Cache-Control", "no-store")
 			}
-			grpcMux.ServeHTTP(resp, req)
+
+			// copy body and return it to request
+			var body []byte
+			if !server.conf.RequestLogging.Disable {
+				body, _ = ioutil.ReadAll(req.Body)
+				req.Body = ioutil.NopCloser(bytes.NewBuffer(body))
+			}
+
+			// handle the request
+			lrw := NewLoggingResponseWriter(resp)
+			grpcMux.ServeHTTP(lrw, req)
+
+			if server.conf.RequestLogging.Disable {
+				return
+			}
+
+			// copy whitelisted headers for logging
+			headers := extractHeaderKeys(req.Header, server.conf.RequestLogging.HeaderWhitelist)
+
+			// log the request
+			entry := log.WithFields(log.Fields{
+				"path":    req.URL.Path,
+				"request": string(body),
+				"header":  headers,
+				"latency": time.Since(start).String(),
+				"status":  lrw.statusCode,
+			})
+			if lrw.statusCode == http.StatusOK {
+				entry.Info("HTTP server responded")
+			} else {
+				entry.Error("HTTP server responded")
```

Comment:
Log the non ok status code?

Reply:
The status code is always logged. This logic just sets the log level.